Create schema.sql

freeridersochi opened the pull request:
Создали структуру данных

freeridersochi committed:
Create schema.sql

zeckson commented on data/scheme.sql:
```diff
@@ -0,0 +1,19 @@
+    name CHAR(10) NOT NULL UNIQUE
+);
+
+insert into marks (name) values('hot'),('sale'),('new');
```
Давай сделаем здесь как и везде в верхнем регистре и отформатируем

zeckson reviewed (changes requested):
Добавил комментариев

freeridersochi committed:
Fixed)

freeridersochi committed:
Продолжил создавать таблицы  и вносить в них данные

Правильно ли я заполнил таблицу photos?
Какой тип данных указывать при ссылке на таблицу? Писать в скобочках название таблицы "(marks)" например или там сложнее механизм?

zeckson commented on data/scheme.sql:
```diff
@@ -0,0 +1,46 @@
+
+INSERT INTO photos ( id, data )
+
+VALUES (0 , '../img/snowboards/1_1.jpg', 
```
так просто не получится, нужно прочитать данные из файла при помощи функции
https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_load-file

zeckson commented on data/scheme.sql:
```diff
@@ -0,0 +1,46 @@
+);
+
+INSERT INTO photos ( id, data )
+
```
давай уберем этот пробел

zeckson commented on data/scheme.sql:
```diff
@@ -0,0 +1,46 @@
+    discount DOUBLE(3,2)
+ 
+/*    
+    mark(),
```
информацию про `FOREIGN_KEY` скину позже немного

freeridersochi committed:
Использовал функцию, лишний пробел

zeckson commented on data/scheme.sql:
```diff
@@ -0,0 +1,46 @@
+ 
+/*    
+    mark(),
+    photo(),
```
здесь надо проставить внешние ключи. Вот здесь можно ознакомиться как они устроены:
https://webformyself.com/sql-vneshnij-klyuch/
https://metanit.com/sql/mysql/2.5.php
https://andreyex.ru/bazy-dannyx/uchebnoe-posobie-po-sql/sql-foreign-key-vneshnij-klyuch/
Если будут вопросы, задавай.

freeridersochi replied:
Да. Возник. 
Из таблицы marks например мне нужно забрать значение id (mark INT references marks (id) ) этого достаточно или мне нужно указать конкретное id  и его значение пусть это будет 2 (new)  чтобы sql  подставил его в таблицу products используя имя марки под этим id?

zeckson replied:
Во-первых, ты всегда можешь попробовать. База данных тебе не даст выполнить противоречивое действие.
Во-вторых, у тебя должно получится что-то вроде:
```
  mark_id INT UNSIGNED NOT NULL,  
  FOREIGN KEY (mark_id) REFERENCES marks (id) ON DELETE RESTRICT,
```

zeckson commented:
Свяжи таблицы, как написано в статьях.
Добавь тестовые данные в таблицу `products`.

freeridersochi committed:
связал таблицы, внес данные

zeckson reviewed (approved):
Отлично!
